src/textbox styler.js

```diff
@@ -0,0 +1,1156 @@
+});
+addDeferredDialogTag('textStyle', function (environment, parameters) {
+	textboxStyler.style(parameters[0]);
+});
+
+// Sets the current Style of the textbox (undefined attributes use Defaults instead)
+// {setTextStyle "StyleName"}
+// {setTextStyleNow "StyleName"}
+addDialogTag('setTextStyleNow', function (environment, parameters, onReturn) {
+	textboxStyler.setStyle(parameters[0]);
+	onReturn(null);
+});
+addDeferredDialogTag('setTextStyle', function (environment, parameters) {
+	textboxStyler.setStyle(parameters[0]);
+});
+
+// Applies the current Style of the textbox (undefined attributes use Defaults instead)
+// {textProperty "StyleProperty, StyleValue"}
+// {textPropertyNow "StyleProperty, StyleValue"}
+addDialogTag('textPropertyNow', function (environment, parameters, onReturn) {
+	var params = parameters[0].split(',');
+	textboxStyler.setProperty(params[0], params[1]);
+	onReturn(null);
+});
+addDeferredDialogTag('textProperty', function (environment, parameters) {
+	var params = parameters[0].split(',');
+	textboxStyler.setProperty(params[0], params[1]);
+});
+
+// Resets the Style of the textbox to the Default style.
+// {resetTextStyle}
+// {resetTextStyleNow}
+addDialogTag('resetTextStyleNow', function (environment, parameters, onReturn) {
+	textboxStyler.resetStyle();
+	onReturn(null);
+});
+addDeferredDialogTag('resetTextStyle', function (environment, parameters) {
+	textboxStyler.resetStyle();
+});
+
+// Resets a Style Property of the textbox to it's Default value.
+// {resetTextProperty "StyleProperty"}
+// {resetTextPropertyNow "StyleProperty"}
+addDialogTag('resetTextPropertyNow', function (environment, parameters, onReturn) {
+	textboxStyler.resetProperty(parameters[0]);
+	onReturn(null);
+});
+addDeferredDialogTag('resetTextProperty', function (environment, parameters) {
+	textboxStyler.resetProperty(parameters[0]);
+});
+
+// Repositions and resizes textbox at an absolute position, based on coordinates.
+// {textPosition "x, y, width, minLines, maxLines"}
+// {textPositionNow "x, y, width, minLines, maxLines"}
+addDialogTag('textPositionNow', function (environment, parameters, onReturn) {
+	var params = parameters[0].split(',');
+	textboxStyler.textboxPosition(params[0],params[1],params[2],params[3],params[4]);
+	onReturn(null);
+});
+addDeferredDialogTag('textPosition', function (environment, parameters) {
+	var params = parameters[0].split(',');
+	textboxStyler.textboxPosition(params[0],params[1],params[2],params[3],params[4]);
+});
+
+export var hackOptions = {
+	// Determines where the textbox is positioned. "shift" moves the textbox based on the player's position.
+	verticalPosition: "shift", // options are "top", "center", "bottom", or "shift" (moves based on player position)
+	horizontalPosition: "center", // options are "left", "center", "right", or "shift" (moves based on player position)
+
+	textboxColor: [0,0,0,255], // Colors text and textbox are drawn, in [R,G,B,A]. TODO: Alpha doesn't presently work!
```

Comment:
guessing this is due to how bitsy's implementation of textbox drawing: it does a `putImageData` instead of drawing the textbox as a texture. the difference is that `put` will directly write pixels to the canvas, vs drawing as a texture being an overdraw

you could look at the transparent dialog hack for an example of how to override it to support alpha